### ISSUE TITLE
"Sending message failed" for specific messages

### ISSUE THREAD

**Issue by vn971** [majn/telegram-purple#305]
Example of a message that cannot be currently sent with telegram-purple:

```
    def render[R <: Rendering](r: R): r.type = r ~~ name ~~ ':' ~~ ' ' ~~ value
```

When trying to send such message to telegram (either personal message or chat), I get no response visually and two error lines in pidgin debug log:

```
(15:17:16) prpl-telegram: Sending message failed.
(15:17:16) prpl-telegram: Sending message failed.
```

**Comment by BenWiederhake** [comment]
That's strange, I can't reproduce that.  What platform and version are you using?  Did you encounter any errors before that?

**Comment by vn971** [comment]
Yes, it happens regularly when I try to paste code in the chat. Note that I'm using the git version of telegram-purple. (Re-compiled and re-checked just now.)

So when you try to send such a message, it is being sent without errors?

**Comment by vn971** [comment]
I'm using ArchLinux, amd64 architecture.

**Comment by EionRobb** [comment]
I had this happen to me (win32) just now when chatting with @BenWiederhake that I tried to send the message "(01:04:07) You feel a disturbance in the force..." after that any messages I sent had the `Sending message failed.` error.  The workaround was to open web.telegram.org and after sending messages through that I was able to send messages via Pidgin again.  Debug below when I couldn't send anything:

```
(01:31:57) util: Writing file prefs.xml to directory C:\Users\Eion\AppData\Roaming\.purple
(01:31:57) util: Writing file C:\Users\Eion\AppData\Roaming\.purple\prefs.xml
(01:31:58) prpl-telegram: update 0x5c486927 (check=0)
(01:31:58) prpl-telegram: tgp_msg_recv before=00000000 server_id=10706
(01:31:58) prpl-telegram: sending all pending recipes
(01:31:58) prpl-telegram: tgl_do_mark_read (115630664)
(01:31:58) prpl-telegram: Sent query #6347605684174889412 of size 24 to DC 3
(01:31:58) prpl-telegram: tgp_msg_process_in_ready, queue size=0
(01:31:58) prpl-telegram: wrote state file: wpts=16161 wqts=0 wseq=10837 wdate=1477916970
(01:31:58) prpl-telegram: wrote secret chat file: 0 chats written.
(01:31:58) prpl-telegram: tgprpl_send_typing()
(01:31:58) prpl-telegram: Sent query #6347605684720901732 of size 24 to DC 3
(01:31:58) prpl-telegram: tgprpl_send_im()
(01:31:59) prpl-telegram: parsed 0 images in messages
(01:31:59) prpl-telegram: tgprpl_send_typing()
(01:31:59) prpl-telegram: Sent query #6347605687244029232 of size 24 to DC 3
(01:31:59) prpl-telegram: Sending message failed.
(01:31:59) prpl-telegram: Sending message failed.
(01:31:59) prpl-telegram:
 
(01:31:59) prpl-telegram: update 0x5c486927 (check=0)
(01:32:01) jabber: Recv (ssl)(1):
```

**Comment by BenWiederhake** [comment]
Not in Eion's screenshot(???): one of my telegram clients apparently sent an "away" status (presumably the web client), but only what seems to be a *bit* later than the error he observed.

I can also regularly, but not controllably, reproduce this.  (I know, in Sep 2 I said otherwise.)  My debug log looks similar.

**Comment by vn971** [comment]
The bug is not present in the dev-1.4.0 branch. That's very cool!
I guess this ticket can be closed now.

**Comment by BenWiederhake** [comment]
We will close this as soon 1.4.0 is released.

**Comment by BenWiederhake** [comment]
Included in 1.3.1.

**Comment by BenWiederhake** [comment]
Not included in 1.3.1, since apparently I never wrote the fix.  Huh.

**Comment by BenWiederhake** [comment]
Should be fixed in [v1.4.1](https://github.com/majn/telegram-purple/releases/tag/v1.4.1).
